timesync/peersync/sync.go

```diff
@@ -191,45 +191,42 @@ func (s *Sync) run() error {
 		timer *time.Timer
 		round uint64
 	)
-	s.log.With().Debug("started sync background worker")
-	defer s.log.With().Debug("exiting sync background worker")
+	s.log.Debug("started sync background worker")
+	defer s.log.Debug("exiting sync background worker")
 	for {
 		prs := s.peers.GetPeers()
 		timeout := s.config.RoundRetryInterval
 		if len(prs) >= s.config.RequiredResponses {
-			s.log.With().Debug("starting time sync round with peers",
-				log.Uint64("round", round),
-				log.Int("peers_count", len(prs)),
-				log.Uint32("errors_count", atomic.LoadUint32(&s.errCnt)),
+			s.log.Debug("starting time sync round with peers",
+				zap.Uint64("round", round),
+				zap.Int("peers_count", len(prs)),
+				zap.Uint32("errors_count", atomic.LoadUint32(&s.errCnt)),
 			)
 			ctx, cancel := context.WithTimeout(s.ctx, s.config.RoundTimeout)
 			offset, err := s.GetOffset(ctx, round, prs)
 			cancel()
 			if err == nil {
 				if offset > s.config.MaxClockOffset || (offset < 0 && -offset > s.config.MaxClockOffset) {
-					s.log.With().Warning("peers offset is larger than max allowed clock difference",
-						log.Uint64("round", round),
-						log.Duration("offset", offset),
-						log.Duration("max_offset", s.config.MaxClockOffset),
+					s.log.Warn("peers offset is larger than max allowed clock difference",
+						zap.Uint64("round", round),
+						zap.Duration("offset", offset),
+						zap.Duration("max_offset", s.config.MaxClockOffset),
 					)
 					if atomic.AddUint32(&s.errCnt, 1) == uint32(s.config.MaxOffsetErrors) {
-						return clockError{
-							err:     ErrPeersNotSynced,
-							details: clockErrorDetails{Drift: offset},
-						}
+						return fmt.Errorf("%w: drift = %v", errPeersNotSynced, offset)
```

Comment:
I literally just removed such a dedicated type in this PR. We don't do this in different places, why do it here?

Reply:
The problem I see with `clockError` is that it is non-specific. The actual error is `ErrPeersNotSynced` and then one has to decode the details based on the inner error. I thought that the offset might be useful for the caller of the method to know so they can maybe adjust their behavior accordingly. If it isn't used outside of logging then we don't need that.

On that node `ErrPeersNotSynced` contains more than just a concise description of the problem. It feels more like a log message than an error. How about shortening the error message to:

```go
var ErrPeersNotSynced = errors.New("timesync: peers are not time synced")
```

?

Reply:
I removed the error struct because it wasn't used in any other way than printing it.

Regarding the error string, I'm not sure, it doesn't hurt to have it longer (with `make sure your system clock is accurate`). It might help somebody.

Reply:
An error usually just describes the issue not how to resolve it (that's either done by error handling directly or logged to the user if it can be done automatically).

Reply:
Well, this advisory could be logged before returning the error. The error itself is just bubbled up to the main.go.

Reply:
I don't see an issue with that 🙂